aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1477 @@
+    use std::signer::address_of;
+    use std::string::String;
+    use std::vector;
+
+    /// The salt used to create a resource account during multisig account creation.
+    /// This is used to avoid conflicts with other modules that also create resource accounts with the same owner
+    /// account.
+    const DOMAIN_SEPARATOR: vector<u8> = b"aptos_framework::multisig_account";
+
+    // Any error codes > 2000 can be thrown as part of transaction prologue.
+    /// Owner list cannot contain the same address more than once.
+    const EDUPLICATE_OWNER: u64 = 1;
+    /// Specified account is not a multisig account.
+    const EACCOUNT_NOT_MULTISIG: u64 = 2002;
+    /// Account executing this operation is not an owner of the multisig account.
+    const ENOT_OWNER: u64 = 2003;
+    /// Transaction payload cannot be empty.
+    const EPAYLOAD_CANNOT_BE_EMPTY: u64 = 4;
+    /// Multisig account must have at least one owner.
+    const ENOT_ENOUGH_OWNERS: u64 = 5;
+    /// Transaction with specified id cannot be found.
+    const ETRANSACTION_NOT_FOUND: u64 = 2006;
+    /// Cannot execute the specified transaction simply via transaction_id as the full payload is not stored on chain.
+    const EPAYLOAD_NOT_STORED: u64 = 7;
+    /// Provided target function does not match the hash stored in the on-chain transaction.
+    const EPAYLOAD_DOES_NOT_MATCH_HASH: u64 = 2008;
+    /// Transaction has not received enough approvals to be executed.
+    const ENOT_ENOUGH_APPROVALS: u64 = 2009;
+    /// Transaction has not received enough rejections to be officially rejected.
+    const ENOT_ENOUGH_REJECTIONS: u64 = 10;
+    /// Number of signatures required must be more than zero and at most the total number of owners.
+    const EINVALID_SIGNATURES_REQUIRED: u64 = 11;
+    /// Payload hash must be exactly 32 bytes (sha3-256).
+    const EINVALID_PAYLOAD_HASH: u64 = 12;
+    /// The multisig account itself cannot be an owner.
+    const EOWNER_CANNOT_BE_MULTISIG_ACCOUNT_ITSELF: u64 = 13;
+    /// Multisig accounts has not been enabled on this current network yet.
+    const EMULTISIG_ACCOUNTS_NOT_ENABLED_YET: u64 = 14;
+
+    /// Represents a multisig account's configurations and transactions.
+    /// This will be stored in the multisig account (created as a resource account separate from any owner accounts).
+    struct MultisigAccount has key {
+        // The list of all owner addresses.
+        owners: vector<address>,
+        // The number of signatures required to pass a transaction (k in k-of-n).
+        signatures_required: u64,
+        // Map from transaction id (incrementing id) to transactions to execute for this multisig account.
+        // Already executed transactions are deleted to save on storage but can always be accessed via events.
+        transactions: Table<u64, MultisigTransaction>,
+        // Last executed or rejected transaction id. Used to enforce in-order executions of proposals.
+        last_executed_transaction_id: u64,
```

Comment:
this feels like sequence number to me, should we align on the same name instead of transaction id? id makes me feel it's a hash or something

Reply:
Makes sense. But what would we name next_pending_transaction_id?

Reply:
last_executed_sequence_number + next_sequence_number?

Reply:
Updated